File: backend/database/schema.go
package database

import "time"

type ResponseHTTP struct {
	Success bool        `json:"success"`
	Data    interface{} `json:"data"`
	Message string      `json:"message"`
}

type Users struct {
	Id       string `json:"id"`
	Name     string `json:"name"`
	Email    string `json:"email"`
	Password string `json:"password"`
}

type Data struct {
	Rid      string    `json:"rid"`
	Date     time.Time `json:"timestamp"`
	Age      int       `json:"age"`    // 0 for 15-25 and 1 for >25, assuming age is in this range
	Gender   int       `json:"gender"` // 0 for female and 1 for male
	FeatureA int       `json:"feature_a"`
	FeatureB int       `json:"feature_b"`
	FeatureC int       `json:"feature_c"`
	FeatureD int       `json:"feature_d"`
	FeatureE int       `json:"feature_e"`
	FeatureF int       `json:"feature_f"`
}

type Views struct {
	Vid       string `json:"vid"`
	Filters   string `json:"filters" db:"json"`
	CreatedAt string `json:"created_at"`
	CreatedBy string `json:"created_by"`
}

File: backend/db_drizzle/db/schema.ts
import {
  pgTable,
  json,
  varchar,
  timestamp,
  unique,
  text,
  integer,
} from "drizzle-orm/pg-core";
import { sql } from "drizzle-orm";

export const users = pgTable("users", {
  id: varchar("id").primaryKey().unique(),
  name: varchar("name").default(""),
  email: varchar("email").unique(),
  password: varchar("password"),
});

export const data = pgTable("data", {
  rid: varchar("rid").primaryKey().unique(),
  date: timestamp("timestamp", { mode: "string" }).default(sql`now()`),
  age: integer("age"), // 0 for 15-25 and 1 for >25, assuming age is in this range
  gender: integer("gender"), // 0 for female and 1 for male
  feature_a: integer("feature_a"),
  feature_b: integer("feature_b"),
  feature_c: integer("feature_c"),
  feature_d: integer("feature_d"),
  feature_e: integer("feature_e"),
  feature_f: integer("feature_f"),
});

export const views = pgTable("views", {
  vid: varchar("vid").primaryKey().unique(),
  filters: json("filters").default({
    ageFilter: null,
    dateFilter: null,
    genderFilter: null,
  }),
  created_by: varchar("created_by").references(() => users.id),
  created_at: timestamp("created_at").default(sql`now()`),
});
